lib/combo/generator.ts

```diff
@@ -36,6 +36,10 @@ export class Generator {
     const buildResult = await build(this.opts);
     const rom = await pack(this.monitor, roms, buildResult, customData, this.opts);
     const { log, hash } = randomize(this.monitor, rom, this.opts);
-    return { rom, log, hash };
+    if (this.opts.settings.generateSpoilerLog) {
```

Comment:
It would be better to add that check in the spoiler log pass directly (returning { log: null })